### PR TITLE
e2e: adds inspect drawer tests

### LINKED ISSUE #23686: Implement E2E tests for Inspect Drawer
### TODO: 
- [x] Inspect Drawer should open when opened from the Panel Menu's Inspect button
  - [x] Correct tab should open when selected from Panel Menu's Inspect submenu
- [x] Inspect Drawer should open from New Panel Editor's Query Inspector button
  - [x] Query tab should be opened by default
- [x] Inspect Drawer's tab should change Inspector's content
- [x] Close button should close the Inspect Drawer
- [x] Expand button should make the Inspector full width

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:
~~Not sure if I should merge this with @s-h-a-d-o-w branch or directly to `master`. Right now this PR merges to @s-h-a-d-o-w branch `explore_e2e_test`. Any suggestions?~~ Setting `master` as `base` and adding @s-h-a-d-o-w as co-author because a lot of the work was done by @s-h-a-d-o-w .

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/grafana/issues/23686

**Special notes for your reviewer**:

### CODE REVIEW

File: e2e/suite1/specs/inspect-drawer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    e2e.getScenarioContext().then(({ lastAddedDashboardUid }) => {
+      e2e.flows.openDashboard(lastAddedDashboardUid);
+    });
+    e2e.pages.Dashboard.Toolbar.toolbarItems('Add panel').click();
+    e2e.pages.AddDashboard.addNewPanel().click();
+
+    e2e.components.DataSource.TestData.QueryTab.scenarioSelect().select('CSV Metric Values');
+
+    e2e.components.BackButton.backArrow().click();
+
```
Comment (torkelo) [inline e2e/suite1/specs/inspect-drawer.spec.ts:22]: Not sure we need this step in this case, now with devenv data sources and dashboards you could open one of the devenv dashboards, for example: http://localhost:3000/d/5SdHCadmz/panel-tests-graph?orgId=1  has some graphs that use csv data already.

Comment (hugohaggmark) [reply e2e/suite1/specs/inspect-drawer.spec.ts:22]: Sure I'll look into that, is the uid in the gdev dashboard always the same?

File: packages/grafana-e2e/src/components/index.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  PanelInspectJSON: componentFactory({
+    selectors: {
+      content: 'Panel inspector JSON content',
+    },
+  }),
+  PanelInspectQuery: componentFactory({
+    selectors: {
+      content: 'Panel inspector Query content',
+    },
+  }),
```
Comment (torkelo) [inline packages/grafana-e2e/src/components/index.ts:48]: I do like the flatter strucuture but maybe these should be nested under a PanelInspector { StatsTab,  DataTab: etc}

**Review by torkelo — COMMENTED** [review]
Awesome stuff! I am a bit unclear on what should go under flows and what is under components etc

File: e2e/suite1/specs/inspect-drawer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    expectDrawerExpandAndContract(viewPortWidth);
+
+    expectDrawerClose();
+
+    expectSubMenuScenario('Data');
+    expectSubMenuScenario('Query');
+    expectSubMenuScenario('Panel JSON', 'JSON');
+
+    e2e.flows.openPanelMenuItem(e2e.flows.PanelMenuItems.Edit);
```
[the remark this answers is not in the document]
Comment (hugohaggmark) [reply e2e/suite1/specs/inspect-drawer.spec.ts:36]: My thinking is that a flow is like a "mini" test that includes several pages/components that you can reuse in your bigger tests. But I like your idea to add actions/flows to pages/components, not sure I'll do that in this PR.

File: packages/grafana-e2e/src/components/index.ts (hunk 2 above)
Comment (hugohaggmark) [reply packages/grafana-e2e/src/components/index.ts:48]: Yes, I'm moving away from those deep nested paths, but I'll see what I can do.

File: e2e/suite1/specs/inspect-drawer.spec.ts (hunk 1 above)
Comment (hugohaggmark) [reply e2e/suite1/specs/inspect-drawer.spec.ts]: Fixed :)

File: packages/grafana-e2e/src/components/index.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      Graph,
+    },
+  },
+  Drawer: {
+    General: componentFactory({
+      selectors: {
+        title: (title: string) => `Panel inspector title ${title}`,
+        expand: 'Panel inspector expand',
+        contract: 'Panel inspector contract',
+        close: 'Panel inspector close',
```
Comment (dprokop) [inline packages/grafana-e2e/src/components/index.ts:24]: Shouldn't these be "Drawer ..." ?

Comment (hugohaggmark) [reply packages/grafana-e2e/src/components/index.ts:24]: Yes, good catch will change!

**Review by dprokop — APPROVED** [review]
💯

**Review by torkelo — APPROVED** [review]
great job!

**Pull request merged**